source/common/http/http1/codec_impl.h

```diff
@@ -41,7 +41,7 @@ class ConnectionImpl;
 /**
  * Base class for HTTP/1.1 request and response encoders.
  */
-class StreamEncoderImpl : public StreamEncoder,
+class StreamEncoderImpl : public virtual StreamEncoder,
```

Comment:
why do we need this to be virtual inheritance now? Maybe comment in the doxygen block, or at least in the PR.

Reply:
This is needed because of the implementation model, where we have a base class that implements StreamEncoder, and then a derived class that implements the base class and also implements the derived interface. I will add a comment here and below, but if it's OK I will do it in my next WIP PR.